guide/src/python-from-rust.md

```diff
@@ -1,511 +1,45 @@
+are met. Its lifetime `'py` is a central part of PyO3's API.
 
-        "#,
-            "house.py",
-            "house",
-        )
-        .unwrap();
+The `Python<'py>` token serves three purposes:
 
-        let house_class = custom_manager.getattr("House").unwrap();
-        let house = house_class.call1(("123 Main Street",)).unwrap();
+* It provides global APIs for the Python interpreter, such as [`py.eval_bound()`][eval] and [`py.import_bound()`][import].
+* It can be passed to functions that require a proof of holding the GIL, such as [`Py::clone_ref`][clone_ref].
+* Its lifetime `'py` is used to bind many of PyO3's types to the Python interpreter, such as [`Bound<'py, T>`][Bound].
 
-        house.call_method0("__enter__").unwrap();
+PyO3's types that are bound to the `'py` lifetime, for example `Bound<'py, T>`, all contain a `Python<'py>` token. This means they have full access to the Python interpreter and offer a complete API for interacting with Python objects.
 
-        let result = py.eval_bound("undefined_variable + 1", None, None);
+Consult [PyO3's API documentation][obtaining-py] to learn how to acquire one of these tokens.
 
-        // If the eval threw an exception we'll pass it through to the context manager.
-        // Otherwise, __exit__  is called with empty arguments (Python "None").
-        match result {
-            Ok(_) => {
-                let none = py.None();
-                house
-                    .call_method1("__exit__", (&none, &none, &none))
-                    .unwrap();
-            }
-            Err(e) => {
-                house
-                    .call_method1("__exit__", (e.get_type_bound(py), e.value_bound(py), e.traceback_bound(py)))
-                    .unwrap();
-            }
-        }
-    })
-}
-```
+### The Global Interpreter Lock
 
-## Handling system signals/interrupts (Ctrl-C)
+Concurrent programming in Python is aided by the Global Interpreter Lock (GIL), which ensures that only one Python thread can use the Python interpreter and its API at the same time. This allows it to be used to synchronize code. See the [`pyo3::sync`] module for synchronization tools PyO3 offers that are based on the GIL's guarantees.
 
-The best way to handle system signals when running Rust code is to periodically call `Python::check_signals` to handle any signals captured by Python's signal handler. See also [the FAQ entry](./faq.md#ctrl-c-doesnt-do-anything-while-my-rust-code-is-executing).
+Non-Python operations (system calls and native Rust code) can unlock the GIL. See [the section on parallelism](parallelism.md) for how to do that using PyO3's API.
 
-Alternatively, set Python's `signal` module to take the default action for a signal:
+## Python's memory model
 
-```rust
-use pyo3::prelude::*;
+Python's memory model differs from Rust's memory model in two key ways:
+- There is no concept of ownership; all Python objects are reference counted
```

Comment:
```suggestion
- There is no concept of ownership; all Python objects are shared usually implemented via reference counted
```

Reply:
Thanks, I added "and" to your suggestion and applied 👍